test/api/handlers_test.go

```diff
@@ -0,0 +1,813 @@
+package api
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"reflect"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/NebulousLabs/skynet-accounts/database"
+	"github.com/NebulousLabs/skynet-accounts/email"
+	"github.com/NebulousLabs/skynet-accounts/skynet"
+	"github.com/NebulousLabs/skynet-accounts/test"
+	"gitlab.com/NebulousLabs/fastrand"
+	"gitlab.com/SkynetLabs/skyd/skymodules"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.sia.tech/siad/crypto"
+)
+
+const (
+	// Helper consts for checking the returned errors.
+	badRequest   = "400 Bad Request"
+	unauthorized = "401 Unauthorized"
+)
+
+type subtest struct {
+	Name string
+	Test func(t *testing.T, at *test.AccountsTester)
```

Comment:
Any reason why these fields are exposed?

Reply:
Not really.